docs/website/docs/developers/general/release-management.md

```diff
@@ -4,61 +4,194 @@ icon: octicons/package-16
 
 # Release management
 
+## :octicons-book-16: Overview
+
+Releases are the process by which we build and deliver software to our users.
+We use a mix of automation and operational practices as part of our continuous
+delivery (CD) efforts.
+
+Releases have the following goals (among others):
+
+* Produce and publish easily installable packages to common package managers
+  for users.
+* Introduce checkpoints with predictable versions around which release notes,
+  testing efforts, and other related activities can align.
+* Improve project and ecosystem velocity and stability.
+
+### :octicons-calendar-16: Stable and nightly releases
+
+In addition to stable releases, we build and publish nightly releases too.
+Nightly releases are less tested and may not include all configurations at all
+times, but they are convenient to install as a preview for what the next stable
+release will contain.
+
+![IREE Architecture](../../assets/images/releases_source_to_stable_dark.svg#gh-dark-mode-only)
+![IREE Architecture](../../assets/images/releases_source_to_stable.svg#gh-light-mode-only)
+
+* Stable releases are published to GitHub releases, are pushed to PyPI, and are
+installable via `pip install` using default options.
+* Nightly releases are published to only GitHub releases as "pre-releases" and
+are installable via `pip install` using non-default options like
+[`--find-links`](https://pip.pypa.io/en/stable/cli/pip_install/#finding-packages)
+and our hosted index page at <https://iree.dev/pip-release-links.html>.
+
+### :material-graph-outline: Projects in scope
+
+The IREE release process currently considers packages in these projects to be
+in scope:
+
+* [iree-org/iree](https://github.com/iree-org/iree)
+* [iree-org/iree-turbine](https://github.com/iree-org/iree-turbine)
+* [nod-ai/shark-ai](https://github.com/nod-ai/shark-ai)
+
+With this dependency graph:
+
+```mermaid
+graph TD
+  accTitle: Dependency graph between in-scope packages
+
+  subgraph iree["iree-org/iree"]
+    iree-base-compiler
+    iree-base-runtime
+    iree-tools-tf
+    iree-tools-tflite
+  end
+
+  subgraph turbine["iree-org/iree-turbine"]
+    iree-turbine
+  end
+
+  iree-base-compiler --> iree-turbine
+  iree-base-runtime --> iree-turbine
+
+  subgraph sharkai["nod-ai/shark-ai"]
+    sharktank
+    shortfin
+    shark-ai
+  end
+
+  iree-base-compiler --> sharktank
+  iree-turbine --> sharktank
+  iree-base-runtime -. source dependency .-> shortfin
+  sharktank -- meta package --> shark-ai
+  shortfin -- meta package --> shark-ai
```

Comment:
```suggestion
  sharktank --> shark-ai (meta package)
  shortfin --> shark-ai (meta package)
```
?

Reply:
Yeah, I'll make a few edits to how shark-ai is represented here.

* It might not always be just a meta package, it might include more code itself
* The other packages have dependencies that don't specify exact versions (or even ranges right now), while shark-ai specifies exact versions

Reply:
Added a new section: https://scotttodd.github.io/iree/developers/general/release-management/#types-of-dependency-links

Reply:
Awesome, I the new section is really great.